### PR TITLE
[SCB-1538] add dependency licenses and better some docs

### PR DESCRIPTION
**Pull request by alec-z**
1.better docs: change Development Guide -> Development guide  and add start server instruction to docs
2.add dependency licenses and change the build_binary.sh to add these licenses to binary package

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -199,3 +199,387 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
+
```
Comment (WillemJiang) [inline LICENSE]: This license is for the source bundle, which is quite different from the binary bundle.
You can find more information here https://www.apache.org/dev/licensing-howto.html#bundled-vs-non-bundled.

Comment (alec-z) [reply LICENSE]: I have changed all the issues you mentioned.

File: licenses/LICENSE [hunk 2]
```diff
@@ -214,27 +214,32 @@ following licenses.
...
 gopkg.in/yaml.v2 (v2.2.1)
-github.com/go-chassis/paas-lager (v1.0.2-0.20190328010332-cf506050ddb2)
 github.com/xdg/scram (v0.0.0-20180814205039-7eeb5667e42c)
 github.com/xdg/stringprep (v1.0.0)
 github.com/go-mesh/openlogging (v1.0.1-0.20181205082104-3d418c478b2d)
 
+================================================================
+The following component(s) are provided under the Apache-2.0 License (Apache License 2.0).
+See the respective project link for details.
+You can find a copy of the License at LICENSE_go-chassis_go-chassis.
```
Comment (WillemJiang) [inline licenses/LICENSE]: Could you rename the Licencese file name to LICENSE_Apache2.0 instead of using LICENSE_go-chassis_go-chassis.

Comment (alec-z) [reply licenses/LICENSE]: changed

**Comment by alec-z** [comment]
@WillemJiang

File: licenses/LICENSE_Apache2.0 [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2019-present, tianxiaoliang
```
Comment (WillemJiang) [inline licenses/LICENSE_Apache2.0]: We need to replace this license with the generic ASF license.

File: licenses/LICENSE [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+github.com/mwitkow/go-conntrack (v0.0.0-20161129095857-cc309e4a2223)
+gopkg.in/yaml.v2 (v2.2.1)
+github.com/xdg/scram (v0.0.0-20180814205039-7eeb5667e42c)
+github.com/xdg/stringprep (v1.0.0)
+github.com/go-mesh/openlogging (v1.0.1-0.20181205082104-3d418c478b2d)
+
+================================================================
+The following component(s) are provided under the Apache-2.0 License (Apache License 2.0).
+See the respective project link for details.
+You can find a copy of the License at LICENSE_Apache2.0 .
```
Comment (WillemJiang) [inline licenses/LICENSE]: It's go-chassis related license, we should keep the name of  go-chassis license.

File: licenses/LICENSE [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+
+The Apache servicecomb-kie project contains subcomponents with
+seperate copyright notices and license terms. Your use of the binary release
+for these subcomponents is subject to the terms and conditions of the
+following licenses.
+
+
+================================================================
+The following component(s) are provided under the Apache-2.0 License (Apache License 2.0).
+See the respective project link for details.
```
Comment (WillemJiang) [inline licenses/LICENSE]: We need to provide the License file link here.

**Comment by alec-z** [comment]
@WillemJiang  LICENSE AND NOTICE changed for go-chassis version upgrade, and travis has passed

File: build/build_binary.sh [hunk 6]
```diff
@@ -82,7 +82,8 @@ EOM
 }
 
 writeConfig
-cp ${PROJECT_DIR}/LICENSE ${PROJECT_DIR}/NOTICE ${release_dir}
+cp ${PROJECT_DIR}/licenses/LICENSE ${PROJECT_DIR}/licenses/NOTICE ${release_dir}
+cp -r ${PROJECT_DIR}/licenses ${release_dir}
```
Comment (WillemJiang) [inline build/build_binary.sh]: So the licenses directory is still have the copied the NOTICE and LICENSE file, we need to clean them up.

File: licenses/LICENSE_Apache2.0 (hunk 3 above)
Comment (WillemJiang) [reply licenses/LICENSE_Apache2.0]: Please clean up this file.

Comment (alec-z) [reply licenses/LICENSE_Apache2.0]: I already told go-chassis to change.

File: licenses/LICENSE (hunk 4 above)
Comment (alec-z) [reply licenses/LICENSE]: go-chassis license is apache-2.0. I change the file name from  LICENSE_go-chassis_go-chassis to LICENSE_Apache2.0 as you told me before. Should I change it back?

File: licenses/LICENSE (hunk 5 above)
Comment (alec-z) [reply licenses/LICENSE]: The follow components are Apache-2.0 License. It's the same as the file content. So I remove the  License file link on purpose.

File: build/build_binary.sh (hunk 6 above)
Comment (alec-z) [reply build/build_binary.sh]: @WillemJiang  Do you mean that we just need to keep the `cp -r ${PROJECT_DIR}/licenses ${release_dir}`?
This is the build_binary.sh. I checked the servicecomb-mesher binary package, which has a LICENSE file  and a licenses directory in the root directory.

Comment (alec-z) [reply build/build_binary.sh]: @WillemJiang  Do you mean that we just need to keep the `cp -r ${PROJECT_DIR}/licenses ${release_dir}`?
This is the build_binary.sh. I checked the servicecomb-mesher binary package, which has a LICENSE file  and a licenses directory in the root directory. 
@WillemJiang  Do you mean that we just need to keep the `cp -r ${PROJECT_DIR}/licenses ${release_dir}`?
This is the build_binary.sh. I checked the servicecomb-mesher binary package, which has a LICENSE file  and a licenses directory in the root directory.

File: licenses/NOTICE [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+=======================================================================
+Apache servicecomb-kie Subcomponents:
+
+
+================================================================
+You can find a copy of the Notice at NOTICE_go-chassis_go-chassis for the the following component(s)
```
[the remark this answers is not in the document]
Comment (alec-z) [reply licenses/NOTICE]: If I clean up "Copyright 2019-present, tianxiaoliang", Should I merge the go-chassis license section to the main apache2.0 license section  and remove the LICENSE_Apache2.0 file?

[the remark this answers is not in the document]
Comment (alec-z) [reply licenses/NOTICE]: Should I keep LICENSE & NOTICE file both in the root directory and the license directory? I Checked mesher and it keep both.

[the remark this answers is not in the document]
Comment (alec-z) [reply licenses/NOTICE]: changed

File: build/build_binary.sh (hunk 6 above)
Comment (WillemJiang) [reply build/build_binary.sh]: yes, we just need to keep the LICENSE and NOTICE file in the root directory of the artifact.

**Comment by WillemJiang** [comment]
@alec-z  Could create a JIRA to track this issue, we need to add the JIRA number on the commit log.

File: licenses/LICENSE [hunk 8]
```diff
@@ -213,6 +213,7 @@ following licenses.
 ================================================================
 The following component(s) are provided under the Apache-2.0 License (Apache License 2.0).
 See the respective project link for details.
+You can find a copy of the License at ./licenses/Apache2.0 .
```
Comment (WillemJiang) [inline licenses/LICENSE]: It should be ./licenses/LICENSE_Apache2.0

**Pull request merged**